tools/codegen/gen_lazy_tensor.py

```diff
@@ -345,17 +358,51 @@ def gen_key(func: FunctionSchema) -> Tuple[str, str]:
             "torch/csrc/lazy/core/shape.h",
             "vector",
         ]],
-        'lazy_ir_inc': [f'#include "{path}"' for path in [
-            node_base_hdr if node_base_hdr is not None else None
-        ] if path is not None],
-        'ir_declarations': list(concat_map_codegen(
-            lazy_ir_cls(backend_indices[backend_key], node_base),
-            grouped_native_functions
-        )),
+        'lazy_ir_inc': [f'#include "{node_base_hdr}"'] if node_base_hdr is not None else [],
+        'ir_declarations': list(concat_map_codegen(lazy_ir_obj, grouped_native_functions)),
         'namespace_prologue': ns_helper.prologue,
         'namespace_epilogue': ns_helper.epilogue,
     })
 
+    # Generate IR Node classes
+    op_headers: List[Optional[str]] = glob.glob(
+        os.path.join(os.path.dirname(node_base_hdr), "*ops*", "*.h")
+    ) if node_base_hdr is not None else []
+
+    fm.write_with_template('LazyIrBuilder.h', 'LazyIrBuilder.h', lambda: {
+        'lazy_ir_builder_inc': [f'#include <{path}>' for path in [
+                "torch/csrc/lazy/core/ir.h",
+                "torch/csrc/lazy/core/ir_nodes.h",
+                "torch/csrc/lazy/core/internal_ops/ltc_ops.h",
+                "torch/csrc/lazy/core/shape_inference.h",
+                node_base_hdr,
+            ] + op_headers
+            if path
+        ],
+        'ir_builder_declarations': dest.generate_lazy_ir_builder(non_native, lazy_ir_obj),
+        'namespace_prologue': ns_helper.prologue,
+        'namespace_epilogue': ns_helper.epilogue,
+    })
+
+    if backend_name == "TorchScript":
```

Comment:
hmm, probably better to use an explicit option in the codegen for enabling special functionality; but let me first think about why this needs to be special in the first place

Reply:
The reason it needs to be special is because the file is being generated to `torch/csrc/lazy/core` which presumably only the TS backend will have access to

Reply:
> The reason it needs to be special is because the file is being generated to `torch/csrc/lazy/core` which presumably only the TS backend will have access to

I would think that all backends have access to stuff in `torch/csrc/lazy/core`, and actually the 'generated' folder is more backend-specific since it gets generated per-backend.  I'm confused about this part.

Reply:
Only the TS backend has "write-access" to `torch/csrc/lazy/core`. In my latest local changes, I did change this to a boolean flag to make the intent clearer